src/Agent.Worker/AgentPluginManager.cs

```diff
@@ -122,6 +103,29 @@ public async Task RunPluginTaskAsync(IExecutionContext context, string plugin, D
             runtimeVariables.CopyInto(pluginContext.Variables, translateToHostPath);
             context.TaskVariables.CopyInto(pluginContext.TaskVariables, translateToHostPath);
 
+            return pluginContext;
+        }
+
+        public async Task RunPluginTaskAsync(IExecutionContext context, string plugin, Dictionary<string, string> inputs, Dictionary<string, string> environment, Variables runtimeVariables, EventHandler<ProcessDataReceivedEventArgs> outputHandler)
+        {
+            ArgUtil.NotNullOrEmpty(plugin, nameof(plugin));
+
+            // Only allow plugins we defined
+            if (!_taskPlugins.Contains(plugin))
+            {
+                throw new NotSupportedException(plugin);
+            }
+
+            // Resolve the working directory.
+            string workingDirectory = HostContext.GetDirectory(WellKnownDirectory.Work);
+            ArgUtil.Directory(workingDirectory, nameof(workingDirectory));
+
+            // Agent.PluginHost
+            string file = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Bin), $"Agent.PluginHost{Util.IOUtil.ExeExtension}");
+            ArgUtil.File(file, $"Agent.PluginHost{Util.IOUtil.ExeExtension}");
+
+            var pluginContext = GeneratePluginExecutionContext(context, inputs, runtimeVariables);
```

Comment:
Is this used anywhere? I'm a little confused what you're doing here

Reply:
I wanted to write a unit test for this logic, but not necessarily kick off the execution of the plugin process so I pulled it out into its own method.

Reply:
Ah, I see - this diff was killing me.